CMakeLists.txt

```diff
@@ -399,6 +404,11 @@ if(ANDROID)
 	SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 	SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 	SET(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+elseif(HAIKU)
+	set(USE_X11 0)
+	set(USE_UPNP 0)
+	set(USE_EGL 0)
+	list(APPEND LIBS network)
```

Comment:
Dolphin itself uses POSIX socket functions in the Core. And the LIBS variable seems to only affect that (see traversal_server which I also had to add a patch for.)